common/test/unit/common/test_file_utils.py

```diff
@@ -29,3 +30,57 @@ def test_calculate_size(self):
             test_file.close()
 
         self.assertEquals(file_size, 1675)
+
+
+class MakePackagesRelativePathTests(unittest.TestCase):
+    def test_PULP_PACKAGES_DIR(self):
+        """Test that PULP_PACKAGES_DIR is in returned path."""
+        self.assertTrue(file_utils.make_packages_relative_path("foo.rpm").startswith(
+            constants.PULP_PACKAGES_DIR))
+
+    def test_lower_letter(self):
+        """Test if folder is created from name starting with lowercase letter."""
+        expected = constants.PULP_PACKAGES_DIR + "/f/foo.rpm"
+        result = file_utils.make_packages_relative_path("foo.rpm")
+        self.assertEqual(expected, result)
+
+    def test_upper_letter(self):
+        """Test if folder is created from name starting with uppercase letter."""
+        expected = constants.PULP_PACKAGES_DIR + "/f/Foo.rpm"
+        result = file_utils.make_packages_relative_path("Foo.rpm")
+        self.assertEqual(expected, result)
+
+    def test_number(self):
+        """Test if folder is created from name starting with number."""
+        expected = constants.PULP_PACKAGES_DIR + "/1/1foo.rpm"
+        result = file_utils.make_packages_relative_path("1foo.rpm")
+        self.assertEqual(expected, result)
+
+    def test_not_basename(self):
+        """Test if correct path is returned when supplied filename is not basename."""
+        expected = constants.PULP_PACKAGES_DIR + "/f/foo.rpm"
+        result = file_utils.make_packages_relative_path("bar/foo.rpm")
+        self.assertEqual(expected, result)
+
+    @mock.patch("os.path.join")
+    def test_join(self, mock_path_join):
```

Comment:
For a function with such well-defined input and output behavior, I don't think you need to test how the code is written. If I rewrote the function to not use ``os.path.join``, but it still had all the correct behavior, the unit tests should all still pass. So I recommend omitting this test.